Entity/Player/g_player.cpp

```diff
@@ -56,19 +79,33 @@ void Player::LoadModel(const char* path, glm::vec3 initialPosition) {
     m_EllipsoidCollider = GetEllipsoidCollider();
 }
 
+// NOTE: This is not being used now as the entity should not own
+// and/or update a camera by itself. The camera is an entity itself.
+// We leave it here anyways, because this shows how it could be done
+// if it turns out that camera-entities are dumb (I don't think so
+// but still...).
 void Player::UpdateCamera(Camera* camera) {
     // Fix camera position
-    camera->m_Pos.x = m_Model.position.x;
-    camera->m_Pos.y = m_Model.position.y;
-    camera->m_Pos.z = m_Model.position.z + 70.0f;
-    camera->m_Pos += (-m_Forward * 80.0f);
-    // m_RotationAngle should already have the information about if we want to move left or right
-    if ( KeyPressed(SDLK_RIGHT) || KeyPressed(SDLK_LEFT) ) {
-        camera->RotateAroundUp(m_RotationAngle);
-    }
+    //camera->m_Pos.x = m_Model.position.x;
+    //camera->m_Pos.y = m_Model.position.y;
+    //camera->m_Pos.z = m_Model.position.z + 70.0f;
+    //camera->m_Pos += (-m_Forward * 80.0f);
+    //// m_RotationAngle should already have the information about if we want to move left or right
+    //if (KeyPressed(SDLK_RIGHT) || KeyPressed(SDLK_LEFT)) {
+    //    camera->RotateAroundUp(m_RotationAngle);
+    //}
 }
```

Comment:
just a thought that popped into my head. the player and the geometry of our fpv should somehow be connected to the camera. right? i heard about rendering the player into a separate buffer so that it always is on top and is not clipping with e.g. walls. but it should somehow know where the position is, no?

ok enough rambling. the question for me is: what is the preferred concept for the player/controller/camera stuff.

Reply:
There is no concept as of now. We will see how it pans out. Many possibilities to do this. We could just have an entity that owns the camera (like the fly cam) and call it a day. And the "controller" is just implemented within the entity. This is how LithTech does it. Interestingly, in LithTech, the Camera is something called an "Engine" Entity, that is, a low level engine component, and the Player Entity defined by the game (non-engine code),  define what is called a "Simulation Object".
Anyways: I am going to implement a simple entity that owns a camera and on input the whole thing can be updated. That is the easiest thing to do right now without having to pass pointers around.